### PR TITLE
[imath] Avoid conflicts between openexr and imath

### PR DESCRIPTION
**Pull request by fabiencastan**
**Describe the pull request**

Imath is included in openexr and the install of imath or openexr will overide the other one.
So, currently the system is broken and we cannot install both openexr and alembic, as it will end up in 2 differents versions mixed up.

- #### What does your PR fix?
Use openexr which contains the latest version of imath.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline]
Does not change anything.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
Not yet, I would like to get feedback.

### CODE REVIEW

**Comment by dg0yt** [comment]
I think this PR is invalid. (The problem is valid, but the solution will be different.)

> Use openexr which contains the latest version of imath.

Actually port `imath` is the latest version.  Port `openexr` in vcpkg is an older version, which includes an older version of imath.

You can install both ports at the same time, but you can't use them at the same time.
For this reason, you currently cannot install `openimageio[opencolorio]` with the latest versions of these ports, because `opencolorio` was updated to use the new `imath` while `openimageio` still uses the outdated `openexr`.
However, in manifest mode, you could request an old, compatible version of `opencolorio`.
There is a version check added as a patch to `openimageio`:
https://github.com/microsoft/vcpkg/blob/55112eb54a1c4cfaec5dd2592a39cea72f78ed56/ports/openimageio/imath-version-guard.patch#L9-L13

The ultimate solution is to update `openexr` to version 3. There have been two attempts already, but they are somewhat stuck due to depending ports not being ready for the update. And vcpgk makes it hard to have an `openexr3` (or `openexr2`) in addition to `openexr`. But you know that, as you commented on these PRs.

**Comment by xchen31u** [comment]
Hi, please accelerate this if possible; is breaking OpenColorIO

File: ports/imath/portfile.cmake [hunk 1]
```diff
@@ -1,24 +0,0 @@
...
-
-vcpkg_cmake_install()
-
-vcpkg_copy_pdbs()
-vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/Imath)
-vcpkg_fixup_pkgconfig()
-
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
-
-file(INSTALL "${SOURCE_PATH}/LICENSE.md" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
Comment (LilyWangLL) [inline ports/imath/portfile.cmake:24]: Please don't just remove this port, add `set(VCPKG_POLICY_EMPTY_PACKAGE enabled)` here, and add `openexr` to `imath`'s dependency in **vcpkg.json**.
You can refer the change of PR #26655.

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by LilyWangLL** [comment]
> I think this PR is invalid. (The problem is valid, but the solution will be different.)
> 
> > Use openexr which contains the latest version of imath.
> 
> Actually port `imath` is the latest version. Port `openexr` in vcpkg is an older version, which includes an older version of imath.
> 
> You can install both ports at the same time, but you can't use them at the same time. For this reason, you currently cannot install `openimageio[opencolorio]` with the latest versions of these ports, because `opencolorio` was updated to use the new `imath` while `openimageio` still uses the outdated `openexr`. However, in manifest mode, you could request an old, compatible version of `opencolorio`. There is a version check added as a patch to `openimageio`:
> 
> https://github.com/microsoft/vcpkg/blob/55112eb54a1c4cfaec5dd2592a39cea72f78ed56/ports/openimageio/imath-version-guard.patch#L9-L13
> 
> The ultimate solution is to update `openexr` to version 3. There have been two attempts already, but they are somewhat stuck due to depending ports not being ready for the update. And vcpgk makes it hard to have an `openexr3` (or `openexr2`) in addition to `openexr`. But you know that, as you commented on these PRs.

Ping @fabiencastan for response.

File: ports/imath/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/imath/portfile.cmake:24]: This part is invalid anyways. Imath is separate from openexr 3.
This whole PR is obsolete with #26862.

**Comment by LilyWangLL** [comment]
Ping @fabiencastan for response. Is work still being done for this PR?

**Comment by dg0yt** [comment]
> Ping @fabiencastan for response. Is work still being done for this PR?

I hope not. Cf. https://github.com/microsoft/vcpkg/pull/26513#discussion_r973912958.

**Comment by fabiencastan** [comment]
No, everything is working fine with the great work to move to openexr-3!

**Pull request closed without merging**